app/models/user.rb

```diff
@@ -412,6 +412,10 @@ def self.find_by_username(username)
     find_by(username_lower: normalize_username(username))
   end
 
+  def in_any_groups?(group_ids)
+    group_ids.include?(Group::AUTO_GROUPS[:everyone]) || (group_ids & groups.map(&:id)).any?
```

Comment:
```suggestion
    group_ids.include?(Group::AUTO_GROUPS[:everyone]) || (group_ids & group_users.pluck(&:group_id)).any?
```

Calling `#groups` will load the AR records which is not really necessary.

Reply:
Sure will strip that back, I will double check that this doesn't cause multiple queries, and if so I will memoize it for the user in a different method e.g. `belonging_to_group_ids` or something